### PR TITLE
Add support for NSX-T Edge Gateway Static Routes

### PR DESCRIPTION
**Pull request by Didainius**
Added support for NSX-T Edge Gateway Static Route configuration via types

* types `NsxtEdgeGatewayStaticRoute`, `types.NsxtEdgeGatewayStaticRoute`
*  methods`NsxtEdgeGateway.CreateStaticRoute`, `NsxtEdgeGateway.GetAllStaticRoutes`, `NsxtEdgeGateway.GetStaticRouteByNetworkCidr`, `NsxtEdgeGateway.GetStaticRouteByName`, `NsxtEdgeGateway.GetStaticRouteById`, `NsxtEdgeGatewayStaticRoute.Update`, `NsxtEdgeGatewayStaticRoute.Delete`

### CODE REVIEW

**Review by lvirbalas — APPROVED** [review]

File: govcd/nsxt_edgegateway_static_route.go [hunk 1]
```diff
@@ -0,0 +1,269 @@
...
+			return nil, fmt.Errorf("error retrieving NSX-T Edge Gateway Static Route after creation: %s", err)
+		}
+
+		var foundStaticRoute bool
+		for _, singleStaticRoute := range allStaticRoutes {
+			if singleStaticRoute.NsxtEdgeGatewayStaticRoute.Name == staticRouteConfig.Name &&
+				singleStaticRoute.NsxtEdgeGatewayStaticRoute.NetworkCidr == staticRouteConfig.NetworkCidr &&
+				singleStaticRoute.NsxtEdgeGatewayStaticRoute.Description == staticRouteConfig.Description {
+				foundStaticRoute = true
+				returnObject = singleStaticRoute
```
Comment (adambarreiro) [inline govcd/nsxt_edgegateway_static_route.go:82]: Should it `break` here? (as you already found a candidate)

File: govcd/nsxt_edgegateway_static_route.go [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+		}
+		err = client.OpenApiGetItem(apiVersion, getUrlRef, nil, returnObject.NsxtEdgeGatewayStaticRoute, nil)
+		if err != nil {
+			return nil, fmt.Errorf("error retrieving NSX-T Edge Gateway Static Route after creation: %s", err)
+		}
+	} else {
+		// ID was not present in response, therefore Static Route needs to be found manually. Using
+		// 'Name', 'Description' and 'NetworkCidr' for finding the entity. Duplicate entries can
+		// exist, but but it should be a good enough combination for finding unique entry until VCD API is fixed
+		allStaticRoutes, err := egw.GetAllStaticRoutes(nil)
```
Comment (adambarreiro) [inline govcd/nsxt_edgegateway_static_route.go:71]: This one doesn't support filtering, right?

**Review by adambarreiro — APPROVED** [review]
LGTM

File: govcd/nsxt_edgegateway_static_route.go (hunk 2 above)
Comment (Didainius) [reply govcd/nsxt_edgegateway_static_route.go:71]: Yeah, sadly the usual stuff.

File: govcd/nsxt_edgegateway_static_route.go (hunk 1 above)
Comment (Didainius) [reply govcd/nsxt_edgegateway_static_route.go:82]: Fixed and ran sdk and terraform tests

**Review by dataclouder — APPROVED** [review]

File: govcd/nsxt_edgegateway_static_route_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+	check.Assert(err, IsNil)
+	nsxtVdc, err := org.GetVDCByName(vcd.config.VCD.Nsxt.Vdc, false)
+	check.Assert(err, IsNil)
+	edge, err := nsxtVdc.GetNsxtEdgeGatewayByName(vcd.config.VCD.Nsxt.EdgeGateway)
+	check.Assert(err, IsNil)
+
+	// Switch Edge Gateway to use dedicated uplink for the time of this test and then turn it off
+	err = switchEdgeGatewayDedication(edge, true) // Turn on Dedicated Tier 0 gateway
+	check.Assert(err, IsNil)
+	defer switchEdgeGatewayDedication(edge, false) // Turn off Dedicated Tier 0 gateway
```
Comment (adezxc) [inline govcd/nsxt_edgegateway_static_route_test.go:24]: Just to be super certain
```suggestion
	defer func() {
	err = switchEdgeGatewayDedication(edge, false)
	check.Assert(err, IsNil)
	} // Turn off Dedicated Tier 0 gateway
```

**Review by adezxc — APPROVED** [review]

File: govcd/nsxt_edgegateway_static_route_test.go (hunk 3 above)
Comment (Didainius) [reply govcd/nsxt_edgegateway_static_route_test.go:24]: Changed

**Review by adezxc — APPROVED** [review]

**Pull request merged**